### DISCUSSION TITLE
How to start rustdesk at boot (before login) on MXLinux or Linux Mint XFCE?

### DISCUSSION THREAD

**Discussion by electricar** [rustdesk/rustdesk#5558] [Q&A]
I am not able to set up rustdesk on my remote PC to start on every boot, so that I can connect remotely to it before the login.
It was working on Linux Mint Cinnamon, but I am not able to set it up on MXLinux or Linux Mint XFCE.
I enabled rustdesk in the "Application Autostart" to start on every "restart", but nothing happens.
Does anyone know hot to do that?
Thank you very much in advance.

**Comment by SergeyMy** [comment]
it must be installed as a system service!

**Reply by C-Duv** [reply]
Is it Linux only or can it also be done on macOS?

**Reply by xlionjuan** [reply]
macOS has another implementation, but if you click start service on RustDesk, it will install itself as service in macOS.

**Reply by C-Duv** [reply]
Following [`src/platform/privileges_scripts/install.scpt`](https://github.com/rustdesk/rustdesk/blob/edb5529df3b505941d8fcb27d818b21b652bcaaf/src/platform/privileges_scripts/install.scpt), I ran:

```
launchctl load -w /Library/LaunchDaemons/com.carriez.RustDesk_service.plist
```

without luck.

I do have `/Applications/RustDesk.app/Contents/MacOS/RustDesk --server` running on boot (without any user logged in) but cannot connect to RustDesk session from another computer.

**Reply by xlionjuan** [reply]
Try latest 1.2.4 nightly

**Reply by C-Duv** [reply]
Same :(
I opened a dedicated discussion: https://github.com/rustdesk/rustdesk/discussions/7762

**Comment by electricar** [comment]
> it must be installed as a system service!
Ok, but how to do it on MXLinux without systemd? I am not familiar with that.

**Comment by SergeyMy** [comment]
> it must be installed as a system service!
> Ok, but how to do it on MXLinux without systemd? I am not familiar with that.

The default is SysVinit, but you can also choose Systemd.

**Comment by xlionjuan** [comment]
If your distro use systemd
`sudo systemctl enable rustdesk.service`

**Reply by electricar** [reply]
I am getting the following error:
![rustdesk](https://github.com/rustdesk/rustdesk/assets/16311580/102de26d-10b2-4fc0-8009-ac5187301c24)

In /usr/lib/rustdesk/ 
I got folders "data" and "lib" and a file rustdesk.
I copied the file rustdesk into the folder /usr/bin/

In /usr/share/rustdesk/files/systemd/ 
I got the file rustdesk.service with this content:

[Unit]
Description=RustDesk
Requires=network.target
After=systemd-user-sessions.service

[Service]
Type=simple
ExecStart=/usr/bin/rustdesk --service
PIDFile=/run/rustdesk.pid
KillMode=mixed
TimeoutStopSec=30
User=root
LimitNOFILE=100000

[Install]
WantedBy=multi-user.target

What is missing here?

**Comment by SergeyMy** [comment]
![изображение](https://github.com/rustdesk/rustdesk/assets/131688106/78605ac4-762d-464a-9f3d-e93f2a090b08)
everything should be there, but
![изображение](https://github.com/rustdesk/rustdesk/assets/131688106/062979be-24a6-4997-9207-13f75d85ac0c)
the binary must have a symlink

![изображение](https://github.com/rustdesk/rustdesk/assets/131688106/2eb7dc9c-5664-4107-b996-84358b54d04f)
all this can be found in the package scripts

**Reply by electricar** [reply]
Thanks, I got it running now after I installed the deb file again after changing my system to systemd.
I got a N95 processor, is it normal that my CPU goes to 30-40% load with a running rustdesk remote session and is at 3% when I do not connect to the PC?
Is it also normal to have multiple rustdesk processes running or did I mess something up?
![rustdesk2](https://github.com/rustdesk/rustdesk/assets/16311580/ea1355a8-302b-47c0-819e-be3036c4540c)

**Comment by SergeyMy** [comment]
any remote session is real-time video stream compression! previously this was not possible, nor a frail processor without hardware acceleration features = very heavy load on the processor.
comfort fee :)

**Comment by SergeyMy** [comment]
not for boasting, for example! connected to itself via relay
![изображение](https://github.com/rustdesk/rustdesk/assets/131688106/754e5dc3-4de0-4acb-af9d-83e51b6ae119)

**Comment by mianmuzammal196** [comment]
Hello, im using the systemd service but the rustdesk service only start after the user login without that nothing, is there a way to start the service on boot not after user login?
EDIT: I set up the service by changing `After=` to `network.target`, but now I'm seeing this error:
<img width="672" height="353" alt="image" src="https://github.com/user-attachments/assets/53ba9483-72f2-4c2d-b4f2-f6e960ba49e8" />

And if I don't log in the user on Ubuntu and accept a screen share, it kills the RustDesk service, and then you are unable to connect again unless you log in to the machine. This makes it very difficult.

**Reply by Harichandrakanthan** [reply]
any solution for this?

**Comment by stdedos** [comment]
@rustdesk any ticket to follow for this? I guess it is one valid requirement to start the service before login
